### PR TITLE
Fix a bug in date hierarchy month/year

### PR DESCRIPTION
**Pull request by idhamari**
Fix ARX issue [366](https://github.com/arx-deidentifier/arx/issues/366).

### CODE REVIEW

File: src/main/org/deidentifier/arx/aggregates/HierarchyBuilderDate.java [hunk 1]
```diff
@@ -534,10 +534,11 @@ private String generalize(String input, Granularity granularity) {
...
             return sdf.format(date);
         } else {
-            int year = Integer.valueOf(sdf.format(date));
-            int lower = (year / _range) * _range;
-            int upper = ((year / _range) + 1) * _range;
-            return "[" + lower + ", " + upper + "[";
+            int dateUnit = Integer.valueOf(sdf.format(date));
+            int lower    = Integer.valueOf((dateUnit - 1) / (_range))  * (_range) + 1;
+            int upper    = lower + _range - 1;
+            String outputDate = "["+lower+"-"+upper+"]";
```
Comment (prasser) [inline src/main/org/deidentifier/arx/aggregates/HierarchyBuilderDate.java:540]: There are blanks missing before and after "+" in this line.

**Review by prasser — CHANGES_REQUESTED** [review]
Looks good apart from minor formatting issues. One question: have you checked wether this also works with Millenia, e.g.? Or are you very confident that the code in lines 537-540 is correct for all cases (not only quarters)?

File: src/main/org/deidentifier/arx/aggregates/HierarchyBuilderDate.java (hunk 1 above)
Comment (idhamari) [reply src/main/org/deidentifier/arx/aggregates/HierarchyBuilderDate.java:540]: > There are blanks missing before and after "+" in this line.

Do you mean line 540? I kept the old line format. If you like I can update it as well. 

> have you checked wether this also works with Millenia, e.g.? Or are you very confident that the code in lines 537-540 is correct for all cases (not only quarters)?

I already tested it on the Month/Year, Quarter a year, Decade, Century, and Millennium cases and it seems to work.

Comment (prasser) [reply src/main/org/deidentifier/arx/aggregates/HierarchyBuilderDate.java:540]: Yes, I mean line 540. And no: you did not keep the old line format ;)

image

PS: Please don't close PRs when changes are requested, but push commit to your branch to adress them.

**Pull request closed without merging**